contributors/design-proposals/node-allocatable.md

```diff
@@ -89,61 +73,253 @@ The flag will be specified as a serialized `ResourceList`, with resources define
+The recommendation is to enforce `SystemReserved` only if a user has profiled their nodes exhaustively to come up with precise estimates.
+
+To begin with enforce `Allocatable` on `pods` only.
+Once adequate monitoring and alerting is in place to track kube daemons, attempt to enforce `KubeReserved` based on heuristics.
+More on this in [Phase 2](#phase-2-enforce-allocatable-on-pods).
+
+The resource requirements of kube system daemons will grow over time as more and more features are added.
+Over time, the project will attempt to bring down utilization, but that is not a priority as of now.
+So expect a drop in `Allocatable` capacity over time.
+
+## Recommended Cgroups Setup
+
+Following is the recommended cgroup configuration for Kubernetes nodes.
+All OS system daemons are expected to be placed under a top level `SystemReserved` cgroup.
+`Kubelet` and `Container Runtime` are expected to be placed under `KubeReserved` cgroup.
+The reason for recommending placing the `Container Runtime` under `KubeReserved` is as follows:
+
+1. A container runtime on Kubernetes nodes is not expected to be used outside of the Kubelet.
+1. It's resource consumption is tied to the number of pods running on a node.
+
+Note that the hierarchy below recommends having dedicated cgroups for kubelet and the runtime to individally track their usage.
+```text
+
+/ (Cgroup Root)
+.
++..systemreserved or system.slice (`SystemReserved` enforced here *optionally* by kubelet)
+.        .    .tasks(sshd,udev,etc)
+.
+.
++..podruntime or podruntime.slice (`KubeReserved` enforced here *optionally* by kubelet)
+.	 .
+.	 +..kubelet
+.	 .   .tasks(kubelet)
+.        .
+.	 +..runtime
+.	     .tasks(docker-engine, containerd)
+.	 
+.
++..kubepods or kubepods.slice (Node Allocatable enforced here by Kubelet)
+.	 .
+.	 +..PodGuaranteed
+.	 .	  .
+.	 .	  +..Container1
+.	 .	  .        .tasks(container processes)
+.	 .	  .
+.	 .        +..PodOverhead
+.	 .        .        .tasks(per-pod processes)
+.	 .        ...
+.	 .
+.	 +..Burstable
+.	 .	  .
+.	 .	  +..PodBurstable
+.	 .	  .	    .
+.	 .	  .	    +..Container1
+.	 .	  .	    .         .tasks(container processes)
+.	 .	  .	    +..Container2
+.	 .	  .	    .         .tasks(container processes)
+.	 .	  .	    .
+.      	 .     	  .    	    ...
+.	 .	  .
+.	 .	  ...
+.	 .
+. 	 .
+.	 +..Besteffort
+.	 .	  .
+.	 .	  +..PodBesteffort
+.	 .	  .    	    .
+.	 .	  .	    +..Container1
+.	 .	  .	    .         .tasks(container processes)
+.	 .	  .	    +..Container2
+.	 .	  .	    .         .tasks(container processes)
+.	 .	  .	    .
+.      	 .     	  .    	    ...
+.	 .	  .
+. 	 .	  ...
+
+```
+
+`systemreserved` & `kubereserved` cgroups are expected to be created by users. If Kubelet is creating cgroups for itself and docker daemon, it will create the `kubereserved` cgroups automatically.
```

Comment:
Can you add text clarifying how users communicate to kubelet the cgroups that were preconfigured?